src/sea_ice_model.jl

```diff
@@ -0,0 +1,117 @@
+using Oceananigans.Fields: TracerFields
+using ClimaSeaIce.SeaIceThermodynamics: external_top_heat_flux
+using Oceananigans: tupleit
+
+struct SeaIceModel{GR, TD, D, CL, TS, U, T, IT, IC, STF, SMS, A} <: AbstractModel{TS}
+    grid :: GR
+    clock :: CL
+    timestepper :: TS
+    # Prognostic State
+    velocities :: U
+    tracers :: T
+    ice_thickness :: IT
+    ice_concentration :: IC
+    # Thermodynamics
+    sea_ice_thermodynamics :: TD
+    # Dynamics
+    sea_ice_dynamics :: D
+    # External boundary conditions
+    external_heat_fluxes :: STF
+    external_momentum_stresses :: SMS
+    # Numerics
+    advection :: A
```

Comment:
seems like this is naturally part of `dynamics`, since without `dynamics` we don't have `advection`?

Reply:
I would contend that advection is a part of the tendencies of thickness and concentration and, for this reason, it should be part of the model. 
Advection happens on the long time-step while dynamics calculate the velocities (either implicitly or on a short time step)

Reply:
What is `advection` if `dynamics == nothing`?

Reply:
will advect the tracers with `velocities == 0` if there if there is no dynamics or maybe with prescribed velocities if we pass `PrescribedVelocities` to the model. 

The tendencies for the tracers (thickness and concentrations) are basically just two terms:
- the transport (advection)
- the thermodynamics

both are part of the tendency calculation, it makes sense to keep them together in the tendency calculation instead of separating them out.

Reply:
interesting. so we can have no dynamics, but non-zero velocities? Does that make sense?

Reply:
I'd say this means that we need a word other than `dynamics`.

Reply:
Indeed, what about maybe calling it `momentum_solver`? Then inside the momentum solver (that can be explicit or implicit) we can have a rheology

Reply:
we've tried to reserve the word "solver" for something the solver elliptic equations which I am not sure applies here

Reply:
This conversation makes me confused about what this does though. Where do we specify coriolis forces, drag forces, and potentially other forces that also enter into the momentum balance?

Reply:
Maybe just leave this property out if we don't really know what it's going to look like